crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -369,6 +371,107 @@ impl Timeline {
         Ok(())
     }
 
+    /// Send a message on a thread.
+    ///
+    /// If the message is also a reply, the sender will be added to the mentions of the reply if
+    /// and only if the event has not been written by the sender.
+    ///
+    /// # Arguments
+    ///
+    /// * `content` - The content of the message
+    ///
+    /// * `replied_to_info` - A wrapper that contains the event ID, sender,
+    ///   content and timestamp of the event to reply to
+    ///
+    /// * `is_reply` - Whether or not the message is a reply inside a thread
+    #[instrument(skip(self, content, replied_to_info))]
+    pub async fn send_thread_reply(
+        &self,
+        content: RoomMessageEventContentWithoutRelation,
+        replied_to_info: RepliedToInfo,
+        is_reply: ReplyWithinThread,
```

Comment:
Instead of duplicating the method, could `send_reply` include a variant that determines whether this is to be used in a thread or not, and for the thread variant have the `ReplyWithinThread` field? It seems it could avoid lots of code duplication here, even though it might make the `Raw` case a bit gnarlier.

If the code becomes too complicated, we can keep separate methods, but just curious if it does make things too complicated or not.

Reply:
Great idea! I have tried to implement this with a new `EnforceThread` enum. I think complexity-wise it's ok and most of the lengthy raw case will go away with the next ruma release.